From 43001241b8480c7957c30fd4ab2c29915df08229 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 25 Sep 2022 21:12:45 +0300
Subject: [PATCH] [SW-0] common: rename fuse spi offset variable

Since slave die fuse spi offset needs to be stored as part of
ROM global structure, this patch renames master die fuse spi offset
to d0_fuse_spi_offset.

Change-Id: I5d752f34b536235ff0c9fc17b690ba07091daad0
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index bade748..3fb803e 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -25,7 +25,7 @@
 	.curr_img = 0,
 	.die_id = 0,
 	.sku = 0,
-	.fuse_spi_offset = 0x0,
+	.d0_fuse_spi_offset = 0x0,
 	.fuse_spi = false,
 };
 
diff --git a/arch/gaudi3/arc/spi_flash.c b/arch/gaudi3/arc/spi_flash.c
index 4102dc3..d50c584 100644
--- a/arch/gaudi3/arc/spi_flash.c
+++ b/arch/gaudi3/arc/spi_flash.c
@@ -30,11 +30,11 @@
 
 	/* read value from bank0 */
 	offset = (word * sizeof(uint32_t));
-	*val = spi_flash_readl(arcboot_rom_global.fuse_spi_offset + offset);
+	*val = spi_flash_readl(arcboot_rom_global.d0_fuse_spi_offset + offset);
 
 	/* read value from bank1 and OR with value from bank0 */
 	offset = ((FUSE_WORDS_PER_BANK + word) * sizeof(uint32_t));
-	*val |= spi_flash_readl(arcboot_rom_global.fuse_spi_offset + offset);
+	*val |= spi_flash_readl(arcboot_rom_global.d0_fuse_spi_offset + offset);
 	return 0;
 }
 
@@ -91,7 +91,7 @@
 	if (idx == HL_FW_IMG_NOT_FOUND)
 		return;
 
-	arcboot_rom_global.fuse_spi_offset = desc.spi_offset;
+	arcboot_rom_global.d0_fuse_spi_offset = desc.spi_offset;
 	arcboot_rom_global.fuse_spi = true;
 
 	if (fuse_read(EFUSE_SYSCFG_OFFSET, &val) == -1)
diff --git a/include/arcboot.h b/include/arcboot.h
index 76504c8..90d6adb 100644
--- a/include/arcboot.h
+++ b/include/arcboot.h
@@ -97,7 +97,7 @@
 	uint8_t curr_img;
 	uint8_t die_id;
 	uint8_t sku; /* 0: Dual die 1: Single die */
-	uint32_t fuse_spi_offset;
+	uint32_t d0_fuse_spi_offset;
 	bool fuse_spi;
 	struct hl_fw_boot_ptrs_ctx boot_ptr_ctx[HL_FW_BOOT_PTR_MAX_IMG];
 	struct hl_fw_sec_info sec_data;
